### PR TITLE
Ignore lib directory

### PR DESCRIPTION
**Pull request by Matt-Butler**
### Summary
Add lib to the .gitignore. 

https://github.com/cerner/terra-core/issues/316

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,9 +51,8 @@
     "lint": "npm run lint:js && npm run lint:scss",
     "lint:js": "eslint --ext .js,.jsx .",
     "lint:scss": "lerna run lint:scss",
-    "prepush": "node scripts/prepush/index.js",
     "pretest": "npm run lint",
-    "postinstall": "npm run bootstrap",
+    "postinstall": "npm run compile && npm run bootstrap",
```
Comment (Matt-Butler) [inline package.json]: This script will compile our code for both travis-ci and our local builds.

File: package.json [hunk 2]
```diff
@@ -78,7 +77,6 @@
     "eslint-plugin-react": "^6.9.0",
     "express": "^4.15.2",
     "glob": "^7.1.1",
-    "husky": "^0.13.3",
```
Comment (Matt-Butler) [inline package.json]: We were using husky for the prepublish script. We no longer need the prepublish script since we are not checking in lib.

**Comment by emilyrohrbough** [comment]
This PR looks like it is removing the prepush script as well, correct? Handling the 2nd and 3rd checkboxes on #316?

**Comment by Matt-Butler** [comment]
> This PR looks like it is removing the prepush script as well, correct? Handling the 2nd and 3rd checkboxes on #316?


Correct, this pr will resolve #316

File: package.json (hunk 1 above)
Comment (bjankord) [inline package.json]: The `postinstall` script doesn't look like its being run on in travis CI.

**Comment by Matt-Butler** [comment]
```
/home/travis/build/cerner/terra-core/packages/terra-table/tests/nightwatch.conf.js
  5:41  error  Unable to resolve path to module '../node_modules/terra-toolkit/lib/nightwatch.json'  import/no-unresolved
````
It looks like my tests are failing because nightwatch.json does not exist in src, only lib. 
https://github.com/cerner/terra-core/tree/master/packages/terra-toolkit/lib
https://github.com/cerner/terra-core/tree/master/packages/terra-toolkit/src

As part of this pr, I will also move nightwatch.json over to src.

File: package.json (hunk 1 above)
Comment (Matt-Butler) [inline package.json]: https://travis-ci.org/cerner/terra-core/builds/237962719#L426

It is being executed, but it collapses in the log.

**Pull request merged**